packages/langchain/src/openai/util.ts

```diff
@@ -0,0 +1,205 @@
+import { AIMessage, BaseMessage, ToolMessage } from '@langchain/core/messages';
+import { ChatResult } from '@langchain/core/outputs';
+import { StructuredTool } from '@langchain/core/tools';
+import type {
+  AzureOpenAiChatCompletionChoice,
+  AzureOpenAiChatCompletionFunction,
+  AzureOpenAiChatCompletionTool,
+  AzureOpenAiChatMessage,
+  AzureOpenAiChatCompletionOutput,
+  AzureOpenAiChatCompletionParameters
+} from '@sap-ai-sdk/foundation-models';
+import { zodToJsonSchema } from 'zod-to-json-schema';
+import { AzureOpenAiChatClient } from './chat.js';
+import {
+  LangChainToolChoice,
+  AzureOpenAiChatCallOptions,
+  ToolChoice
+} from './types.js';
+
+/**
+ * Maps a LangChain {@link StructuredTool} to {@link AzureOpenAiChatCompletionFunction}.
+ * @param tool - Base class for tools that accept input of any shape defined by a Zod schema.
+ * @returns The OpenAI chat completion function.
+ */
+function mapToolToOpenAiFunction(
+  tool: StructuredTool
+): AzureOpenAiChatCompletionFunction {
+  return {
+    name: tool.name,
+    description: tool.description,
+    parameters: zodToJsonSchema(tool.schema)
```

Comment:
Step 1: Here `tool` has the property `schema`. This function is called in line 40 `mapToolToOpenAiTool`